src/main/java/roboy/dialog/states/ordinaryStates/GamingTwentyQuestionsState.java

```diff
@@ -0,0 +1,263 @@
+import java.util.List;
+
+public class GamingTwentyQuestionsState extends State {
+
+
+	private static final double PROBABILITY_THRESHOLD = 0.85;
+	private final static String TRANSITION_GAME_ENDED = "gameEnded";
+
+	private Akiwrapper aw = new AkiwrapperBuilder().setFilterProfanity(true).build();
+	private Question nextQuestion = null;
+	private Guess currentGuess = null;
+
+	private List<Long> declined = new ArrayList<>();
+	private boolean userReady = false;
+	private boolean guessesAvailable = false;
+	private boolean gameFinished = false;
+	private int numberGuesses = 0;
+
+	private boolean filterApplied = false;
+	private String winner = "roboy";
+
+	public GamingTwentyQuestionsState(String stateIdentifier, StateParameters params) {
+		super(stateIdentifier, params);
+	}
+
+	@Override
+	public Output act() {
+
+		if(!userReady){
+			return Output.say("Now think of a character and tell me when you're ready to start the game.");
+
+		} else if(guessesAvailable) {
+
+			try {
+				return doAGuess();
+			}
+			catch (Exception e){
+				return Output.say("I have a problem finding a guess...");
+			}
+		}
+		else {
+
+			return askNextQuestion();
+		}
+
+	}
+
+	@Override
+	public Output react(Interpretation input) {
+
+
+		String intent = getIntent (input);
+
+		if(!isUserReady(intent)){
+
+			return Output.say("Nice, then let's start");
+
+		} else if(guessesAvailable){
+
+			return processUserGuessAnswer(intent);
+
+		} else {
+
+			try {
+
+				return saveUsersEstimate(intent);
+			}
+			catch (Exception e){
+				return Output.say("I have a problem processing your input...");
+			}
+		}
+	}
+
+	@Override
+	public State getNextState() {
+
+		if(gameFinished){
+			applyFilter(winner);
+			resetGame();
+			return getTransition(TRANSITION_GAME_ENDED);
+
+		} else {
+			return this;
+		}
+	}
+
+	private String getIntent(Interpretation input) {
+
+
+		List<String	> tokens = input.getTokens();
+
+
+
+
+
+		String intent = null;
+
+		for(String token : tokens) {
+			if(token.equals("yes")) {
+				intent = "yes";
+			} else if (token.equals("no")) {
+				intent = "no";
+			} else if(token.equals("ready")) {
+				intent = "ready";
+			} else if(token.equals("back")) {
+				intent = "back";
+			} else if(token.equals("dont")) {
+				intent = "dont know";
+			} else if(token.equals("not")) {
+				intent = "probably not";
+			} else if(token.equals("probably")) {
+				intent = "probably";
+			}
+		}
+
+		return intent;
+	}
+
+	private Output askNextQuestion() throws NullPointerException{
+
+		String nextQuestionString = null;
+		try {
+			nextQuestion = aw.getCurrentQuestion();
+			nextQuestionString = nextQuestion.getQuestion();
+			++numberGuesses;
+		}
+		catch(NullPointerException e){
+			gameFinished = true;
+			return Output.say("I throw in the towel, I think I have no idea what to ask next. You win!");
+		}
+		return Output.say(nextQuestionString);
+	}
+
+	private Output saveUsersEstimate(String intent) throws Exception {
```

Comment:
avoid adding throws Exception in function definition and handle them inside your function